nuget/lib/dependabot/nuget/file_fetcher.rb

```diff
@@ -42,16 +42,18 @@ def self.required_files_message
         ).void
       end
       def initialize(source:, credentials:, repo_contents_path: nil, options: {})
-        super(source: source, credentials: credentials, repo_contents_path: repo_contents_path, options: options)
+        super
 
         @sln_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
         @sln_project_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
         @project_files = T.let([], T::Array[Dependabot::DependencyFile])
         @fetched_files = T.let({}, T::Hash[String, T::Array[Dependabot::DependencyFile]])
         @nuget_config_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
         @packages_config_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
+        @assembly_binding_redirect_config_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
       end
 
+      # rubocop:disable Metrics/AbcSize
```

Comment:
I wasn't sure how to resolve this easily without making the code worse, so disabled Metrics/AbcSize here.
Let me know if there is an obvious solution I'm missing, not a Ruby expert.

```
lib/dependabot/nuget/file_fetcher.rb:57:7: C: Metrics/AbcSize: Assignment Branch Condition size for fetch_files is too high. [<15, 31, 8> 35.36/35]
      def fetch_files ...
      ^^^^^^^^^^^^^^^
```

Reply:
Look at: https://github.com/dependabot/dependabot-core/pull/9678/files#diff-56cacc3dfbb2aaf55bc204f85d0c923c7fd5b23e908ca8ccfaf455407f5ff509R56-R68